### DISCUSSION TITLE
Need help getting npm to persist in docker code-server

### DISCUSSION THREAD

**Discussion by mia-riezebos** [coder/code-server#4276] [Q&A] [answered]
Using the code-server terminal, one can install nodejs and npm, within the docker container. 
However, after an update or recreation of the container, this installation will be gone as it's not part of the docker image. 
Is there a way to mount these as a volume on the host OR better yet, use the existing nodejs/npm installations from the host within the container?

Thank you!

**Comment by jsjoeio** [comment, marked as answer]
Fantastic question! So when you're running code-server inside a Docker container and install dependencies or packages, they won't persist (as you've explained). Instead, you'll need to install them in the `Dockerfile`. Here's an [example for installing Node](https://github.com/jsjoeio/joe-image/blob/main/Dockerfile#L51-L59).

Here is [another example](https://github.com/cdr/deploy-code-server/blob/main/Dockerfile) which starts from the code-server image.

Let me know if that helps!

**Reply by jsjoeio** [reply]
cc @bpmct - just thought you would appreciate seeing this discussion!
